tests/console/zypper_lifecycle.pm

```diff
@@ -33,17 +33,29 @@ use strict;
 use warnings;
 use testapi;
 use utils;
-use version_utils qw(is_sle is_jeos);
+use version_utils qw(is_sle is_jeos is_upgrade);
+use Utils::Architectures;
 
 our $date_re = qr/[0-9]{4}-[0-9]{2}-[0-9]{2}/;
 
 sub run {
     diag('fate#320597: Introduce \'zypper lifecycle\' to provide information about life cycle of individual products and packages');
+
+    # Workaround for bsc#1166549, zypper lifecycle error with 'Error building the cache'.
+    # Currently the only workaround option is to run zypper-lifecycle as root.
+
+    # Now we just hit bsc#1166549 on s390x and ppc64le, so we just need root console
+    # on these two platforms.
+    my $needs_root_console = (check_var('ARCH', 's390x') || check_var('ARCH', 'ppc64le'));
```

Comment:
this could be combined with the next but it is also ok like it is

Reply:
so what kind of change do you suggest by "this could be combined with the next" ? I can change it in the last minute. thanks

Reply:
I mean the condition is only depending on `arch = /s390x|ppc64le/` because of the bug, same as the if-condition in the next line. So I was thinking of the following:

```perl
my $needs_root_console = 0;
if ((is_s390x || is_ppc64le) && is_sle('>=15-SP3') && is_upgrade) {
    $needs_root_console = 1;
    record_soft_failure 'bsc#1166549 - zypper lifecycle failed to create metadata cache directory';
}
```

Reply:
changed, thanks